bot/cogs/geo_cog/streetviewrandomizer/street_view_static_api.py

```diff
@@ -0,0 +1,94 @@
+import aiohttp as aiohttp
+import asyncio
+import bot.bot_secrets as bot_secrets
+import json
+import logging
+import nest_asyncio
+from bot.cogs.geo_cog.streetviewrandomizer.coordinate import Coordinate
+from timeit import default_timer as timer
+
+ENDPOINT = "https://maps.googleapis.com/maps/api/streetview"
+SESSION = aiohttp.ClientSession()
+nest_asyncio.apply()
+
+
+class StreetViewStaticApi:
+    async def close_session(self):
+        if not self.session.closed:
+            await self.session.close()
+
+    @staticmethod
+    async def geolocate(quota: int, pic_base: str, filename: str, location_params: dict) -> tuple[str, float]:
+        start = timer()
+        if quota > 0:
+            curr_session = aiohttp.ClientSession()
+            await asyncio.sleep(1)
+            async with curr_session.get(url=pic_base,
+                                        params=location_params,
+                                        allow_redirects=False) as resp:
+                pic_response = await resp.read()
+                await curr_session.close()
+                await asyncio.sleep(1)
+                f = open(f'bot/cogs/geo_cog/temp_assets/{filename}', 'wb')
+                f.write(pic_response)
+                f.close()
+
+                end = timer()
+                api_rest_time = (end - start) * 1000
+                return filename, api_rest_time
+
+    @staticmethod
+    async def has_image(coord: Coordinate, radius_m: int) -> tuple[Coordinate, bool]:
+        # Check if the API key exists.
+        if bot_secrets.secrets.geocode_key is None:
+            raise Exception("API key is required. Use --api-key or set the GOOGLE_MAPS_API_KEY environment variable.")
```

Comment:
No need for this check and raise since accessing it in the first place will raise an exception if it's not already configured.